src/utilities/data-formatter.ts

```diff
@@ -223,32 +226,38 @@ var dataFormatter = {
   }
 }
 
-function validateAndCast(value) {
+function validateAndCast(value: string | number | string[] | number[]): string | string[] {
   if (typeof value === 'string') return value;
   if (typeof value === 'number') return value.toString();
-  if (typeof value[0] === 'string') return value;
+  if (typeof value[0] === 'string') return value as string[];
   if (typeof value[0] === 'number') {
-    return value.map((value) => value.toString());
+    return (value as number[]).map((v: number) => v.toString());
   }
   throw new Error(
     'unrecognized value, expected string[], string, number[], or number'
   );
 }
 
+
+/**
+ * TODO: Tighten formatDataIn type once we convert caller to TS
+ */
+type DefaultRNNFormatterThis = { dataFormatter?: DataFormatter, formatDataIn: Function };
+
 /**
  *
  * @param {*[]} data
  * @returns {Number[]}
  */
-function defaultRNNFormatter(data) {
+export function defaultRNNFormatter(this: DefaultRNNFormatterThis, data: any[]): number[] {
```

Comment:
Do we need a `this` here?  Since this is non-extendable, we can enforce `data` type, which is `string` here.

Reply:
'data' is not a string, or at least that is not what the function body is treating it as, for exemple:
` Array.isArray(data[0]`

The function is referencing 'this' in a few places, so we need to type it, not sure what you mean by 'non-extendable'. this function is used as a "method" since it is set to an object property and then invoked as a property of the object (which sets the 'this' keyword to the object) and in testing it is invoked using 'call' to set 'this' to a mock object.

Reply:
Ah, ty.